File: spark/learning-spark-lightning-fast-big-data-analysis/jvm/src/main/scala/redocpot/spark/Union.scala
package scala.redocpot.spark

import org.apache.spark.SparkConf
import org.apache.spark.SparkContext
import org.apache.spark.SparkContext._

object Union {
  def main(args : Array[String]) : Unit = {
    val conf = new SparkConf().setAppName("Union").setMaster("local[*]")
    val sc = new SparkContext(conf)
    val textFile = sc.textFile("/Users/me/Code/spark-learning/learning-spark-lightning-fast-big-data-analysis/pom.xml")
    val xmlLines = textFile.filter(line => line.contains("xml"))
    val mavenLines = textFile.filter(line => line.contains("maven"))
    val unionLines = xmlLines.union(mavenLines)
    unionLines.foreach(line => {
      println(line)
    })
    sc.stop
  }
}

File: spark/learning-spark-lightning-fast-big-data-analysis/jvm/src/main/java/redocpot/spark/Union.java
package redocpot.spark;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

public class Union {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf()
            .setMaster("local")
            .setAppName("Union");
        JavaSparkContext sc = new JavaSparkContext(conf);

        JavaRDD<String> textFile = sc.textFile(
                "/Users/me/Code/spark-learning/learning-spark-lightning-fast-big-data-analysis/pom.xml");
        JavaRDD<String> xmlLines = textFile.filter(line -> line.contains("xml"));
        JavaRDD<String> mavenLines = textFile.filter(line -> line.contains("maven"));
        JavaRDD<String> unionLines = xmlLines.union(mavenLines);
        unionLines.foreach(line -> {
            System.out.println(line);
        });

        sc.stop();
    }
}
